lib/travis/logs/drain_consumer.rb

```diff
@@ -63,12 +63,24 @@ def dead?
       end
 
       private def jobs_queue
+        return jobs_queue_sharded if logs_config[:drain_rabbitmq_sharding]
+        jobs_queue_single
+      end
+
+      private def jobs_queue_single
         @jobs_queue ||= jobs_channel.queue(
           "reporting.jobs.#{reporting_jobs_queue}",
           durable: true, exclusive: false
         )
       end
 
+      private def jobs_queue_sharded
+        @jobs_queue ||= jobs_channel.queue(
+          "reporting.jobs.#{reporting_jobs_queue}",
```

Comment:
I've just realized that this still uses the old queue, which i don't think it should. If we create a new exchange (and pseudo-queue) it would probably makes things easy if they were different.

Reply:
The switch happens in `lib/travis/logs/drain.rb` when creating the consumer. It's quite subtle though, I would be in favour of moving that logic into the drain consumer and hard-coding "logs" vs "logs_sharded" there.

Unless there is some other piece missing somewhere?

Reply:
Ah, I was silly looking at an outdated version of `drain.rb` 😅 . Still, I think it might be less confusing if we drop the `reporting.jobs` bit for the sharded queue, since we won't be using the topic exchange (although that can wait until the worker changes are finalized)

Reply:
Strongly in favour 👍